internal/provider/utils.go

```diff
@@ -1,9 +1,45 @@
 package provider
 
 import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"slices"
+
 	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
 )
 
 func IsValueProvided(value attr.Value) bool {
 	return !value.IsNull() && !value.IsUnknown()
 }
+
+func ValidateResponse(resp *http.Response, body []byte, err error, expected_statuses []int) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	if err != nil {
+		diags.AddError(
+			"Client request error",
+			err.Error(),
+		)
+		return diags
+	}
+	if resp == nil {
+		diags.AddError("HTTP response error", "No HTTP response from server")
+		return diags
+	}
+	if !slices.Contains(expected_statuses, resp.StatusCode) {
+		var info map[string]interface{}
+		err := json.Unmarshal(body, &info)
+		if err != nil {
+			diags.AddError("Error unmarshaling JSON response body", err.Error())
+		}
```

Comment:
You are trying to unmarshal the body which is not expected to be used anywhere.
Is this part of the ValidateResponse flow? in this case, it should be put outside of this if statement

Reply:
The response body is included in the diagnostic detail to give more info on why the unexpected status code was received. But I like Mike's suggestion and will update this function to return the body so we're only unmarshalling it once.

Reply:
Scratch that, see comment below. I think we can probably ignore this particular error though. If we can't unmarshal the response body here then the diagnostics error simply won't include the body details.